### PR TITLE
[KOGITO-4535] - New data index image for Postgresql variation

### PR DESCRIPTION
**Pull request by spolti**
Signed-off-by: spolti <fspolti@redhat.com>
See: https://issues.redhat.com/browse/KOGITO-4535

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors guide](README.md#contributing-to-kogito-images-repository)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a testcase that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -568,7 +571,7 @@ $ docker run -it --env QUARKUS_POSTGRESQL_CONNECTION_STRING=postgresql://localho
 To enable debug just use this env while running this image:
 
 ```bash
-docker run -it --env SCRIPT_DEBUG=true --env QUARKUS_INFINISPAN_CLIENT_SERVER_LIST=my-infinispan-server:11222 quay.io/kiegroup/kogito-data-index-infinispan:latest
+docker run -it --env JAVA_OPTIONS="-Dquarkus.datasource.jdbc.url=jdbc:postgresql://localhost:5432/quarkus -Dquarkus.datasource.username=kogito -Dquarkus.datasource.password=secret" quay.io/kiegroup/kogito-data-index-infinispan:latest
```
Comment (radtriste) [inline README.md]: why postgresql information with infinispan image ?

File: tests/features/common.feature [hunk 2]
```diff
@@ -1,4 +1,13 @@
-@quay.io/kiegroup/kogito-builder @quay.io/kiegroup/kogito-runtime-jvm @quay.io/kiegroup/kogito-runtime-native @quay.io/kiegroup/kogito-data-index-infinispan @quay.io/kiegroup/kogito-data-index-mongodb @quay.io/kiegroup/kogito-trusty-infinispan @quay.io/kiegroup/kogito-trusty-redis @rhpam-7/rhpam-kogito-runtime-jvm-rhel8 @rhpam-7/rhpam-kogito-builder-rhel8
+@quay.io/kiegroup/kogito-builder
+@quay.io/kiegroup/kogito-runtime-jvm
+@quay.io/kiegroup/kogito-runtime-native
+@quay.io/kiegroup/kogito-data-index-infinispan
+@quay.io/kiegroup/kogito-data-index-mongodb
+@quay.io/kiegroup/kogito-data-index-postgresql
+@quay.io/kiegroup/kogito-trusty-infinispan
```
Comment (radtriste) [inline tests/features/common.feature:7]: just noticing we are missing some images there or ? like the jobs-service ...

File: tests/features/data-index/kogito-data-index-postgresql.feature [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  Scenario: verify if the indexing service binaries are available on /home/kogito/bin
+    When container is started with command bash
+    Then run sh -c 'ls /home/kogito/bin/data-index-service-postgresql-runner.jar' in container and immediately check its output for /home/kogito/bin/data-index-service-postgresql-runner.jar
+
+  Scenario: verify if of container is correctly started with postgresql parameters
+    When container is started with env
+      | variable      | value         |
+      | SCRIPT_DEBUG  | true          |
+      | JAVA_OPTIONS  | -Dquarkus.datasource.jdbc.url=jdbc:postgresql://localhost:5432/quarkus -Dquarkus.datasource.username=kogito -Dquarkus.datasource.password=secret |
```
Comment (radtriste) [inline tests/features/data-index/kogito-data-index-postgresql.feature]: do we need to set the username/password for the tests ? 
can we sync with jobs-service postgresql test ? https://github.com/kiegroup/kogito-images/pull/529/files?file-filters%5B%5D=.feature&file-filters%5B%5D=.md&file-filters%5B%5D=.py&file-filters%5B%5D=.yaml#diff-926802640217e0bbc0cfb70dd5f6bc4957ce30839675cb4a7aa7fdcec1b3491eR18

File: README.md (hunk 1 above)
Comment (spolti) [reply README.md]: I don't know lol, thanks for point this out.

File: tests/features/common.feature (hunk 2 above)
Comment (spolti) [reply tests/features/common.feature:7]: good catch, let me add them.

File: tests/features/data-index/kogito-data-index-postgresql.feature (hunk 3 above)
Comment (spolti) [reply tests/features/data-index/kogito-data-index-postgresql.feature]: applied.

File: tests/features/data-index/kogito-data-index-postgresql.feature [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+  Scenario: verify if the indexing service binaries are available on /home/kogito/bin
+    When container is started with command bash
+    Then run sh -c 'ls /home/kogito/bin/data-index-service-postgresql-runner.jar' in container and immediately check its output for /home/kogito/bin/data-index-service-postgresql-runner.jar
+
+  Scenario: verify if of container is correctly started with postgresql parameters
+    When container is started with env
+      | variable      | value                                                                                                  |
+      | SCRIPT_DEBUG  | true                                                                                                   |
+      | JAVA_OPTIONS  | -Dquarkus.datasource.jdbc.url=jdbc:postgresql://localhost:5432/quarkus?username=kogito&password=secret |
```
Comment (radtriste) [inline tests/features/data-index/kogito-data-index-postgresql.feature]: Why do we have `JAVA_OPTIONS` here where in jobs-service we have `QUARKUS_POSTGRES_CONNECTION_STRING`. any real difference ?
This is quite confusing not having the same behavior

Comment (spolti) [reply tests/features/data-index/kogito-data-index-postgresql.feature]: I have left a comment on gchat about it, envs are not working.
@cristianonicolai f.y.i.

Comment (radtriste) [reply tests/features/data-index/kogito-data-index-postgresql.feature]: This is weird. Why would that work on jobs-service and not on data-index ?

File: modules/kogito-data-index-postgresql/module.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+schema_version: 1
+name: org.kie.kogito.dataindex.postgresql
+version: "2.0.0-snapshot"
+
+artifacts:
+  - name: data-index-service-postgresql-runner.jar
+    url: https://origin-repository.jboss.org/nexus/content/groups/public/org/kie/kogito/data-index-service-postgresql/2.0.0-SNAPSHOT/data-index-service-postgresql-2.0.0-20210617.033050-3-runner.jar
```
Comment (vaibhavjainwiz) [inline modules/kogito-data-index-postgresql/module.yaml]: we should use `https://repository.jboss.org/`

Comment (spolti) [reply modules/kogito-data-index-postgresql/module.yaml]: yes, good point, thx.

File: tests/features/data-index/kogito-data-index-postgresql.feature (hunk 4 above)
Comment (spolti) [reply tests/features/data-index/kogito-data-index-postgresql.feature]: addressed.

File: README.md [hunk 6]
```diff
@@ -562,13 +565,13 @@ $ docker run -it --env QUARKUS_MONGODB_CONNECTION_STRING=mongodb://localhost:270
 
 Basic usage with Postgresql:
 ```bash
-$ docker run -it --env QUARKUS_POSTGRESQL_CONNECTION_STRING=postgresql://localhost:5432 quay.io/kiegroup/kogito-data-index-postgresql:latest
+$ docker run -it --env QUARKUS_DATASOURCE_JDBC_URL="jdbc:postgresql://localhost:5432/quarkus -Dquarkus.datasource.username=kogito -Dquarkus.datasource.password=secret" quay.io/kiegroup/kogito-data-index-postgresql:latest
```
Comment (radtriste) [inline README.md]:
```suggestion
$ docker run -it --env QUARKUS_DATASOURCE_JDBC_URL="jdbc:postgresql://localhost:5432/quarkus"  --env QUARKUS_DATASOURCE_USERNAME="kogito" --env QUARKUS_DATASOURCE_PASSWORD="secret" quay.io/kiegroup/kogito-data-index-postgresql:latest
```

as far as I understood, we need to set both jdbc and reactive urls, aka `QUARKUS_DATASOURCE_REACTIVE_URL`

File: tests/features/data-index/kogito-data-index-postgresql.feature [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+  Scenario: verify if the indexing service binaries are available on /home/kogito/bin
+    When container is started with command bash
+    Then run sh -c 'ls /home/kogito/bin/data-index-service-postgresql-runner.jar' in container and immediately check its output for /home/kogito/bin/data-index-service-postgresql-runner.jar
+
+  Scenario: verify if of container is correctly started with postgresql parameters
+    When container is started with env
+      | variable                     | value                                                                    |
+      | SCRIPT_DEBUG                 | true                                                                     |
+      | QUARKUS_DATASOURCE_JDBC_URL  | jdbc:postgresql://10.1.1.10:5432/quarkus?username=kogito&password=secret |
```
Comment (radtriste) [inline tests/features/data-index/kogito-data-index-postgresql.feature]: could we setup username and password as different env vars ?

File: tests/features/jobs-service/kogito-jobs-service-postgresql.feature [hunk 8]
```diff
@@ -17,9 +17,9 @@ Feature: Kogito-jobs-service-postgresql feature.
 
   Scenario: verify if of container is correctly started with postgresql parameters
     When container is started with env
-      | variable                          | value                                          |
-      | SCRIPT_DEBUG                      | true                                           |
-      | QUARKUS_POSTGRES_CONNECTION_STRING | postgres://user:password@localhost:5432/admin  |
+      | variable                           | value                                          |
+      | SCRIPT_DEBUG                       | true                                           |
+      | QUARKUS_POSTGRES_CONNECTION_STRING | jdbc:postgresql://localhost:5432/quarkus?username=kogito&password=test  |
```
Comment (radtriste) [inline tests/features/jobs-service/kogito-jobs-service-postgresql.feature]: Could we use here `QUARKUS_DATASOURCE_*` env vars ?

**Review by radtriste — CHANGES_REQUESTED** [review]

File: tests/features/jobs-service/kogito-jobs-service-postgresql.feature (hunk 8 above)
Comment (spolti) [reply tests/features/jobs-service/kogito-jobs-service-postgresql.feature]: Didn't work well...

```
$ drun --env SCRIPT_DEBUG=true --env QUARKUS_DATASOURCE_JDBC_URL="jdbc:postgresql://172.17.0.2:5432/quarkus" --env QUARKUS_DATASOURCE_USERNAME=kogito --env QUARKUS_DATASOURCE_PASSWORD=secret  quay.io/kiegroup/kogito-jobs-service-postgresql:2.0.0-snapshot

...

2021-06-25 15:42:17,430 db3698ae998e ERROR [org.kie.kogito.jobs.service.scheduler.JobSchedulerManager:-1] (executor-thread-1) Error Loading scheduled jobs!: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: **localhost/127.0.0.1:5432**
Caused by: java.net.ConnectException: Connection refused

File: README.md (hunk 6 above)
Comment (spolti) [reply README.md]: using only QUARKUS_DATASOURCE_JDBC_URL works. no need the react one.

File: tests/features/jobs-service/kogito-jobs-service-postgresql.feature (hunk 8 above)
Comment (spolti) [reply tests/features/jobs-service/kogito-jobs-service-postgresql.feature]: funny thing, with latest artifact even QUARKUS_POSTGRES_CONNECTION_STRING is working

Comment (spolti) [reply tests/features/jobs-service/kogito-jobs-service-postgresql.feature]: nevermind, rebuilding it with the today's artifact version has fixed the QUARKUS_DATASOURCE.... envs

**Review by vaibhavjainwiz — APPROVED** [review]

**Pull request merged**